### PR TITLE
[hatohol-db-initiator][hap2] Replace server type SQL file

### PR DESCRIPTION
**Pull request by Mnakagawa**
And remove create-db.sql from spec file.
Related: #1415 , #2014

### CODE REVIEW

**Comment by cosmo0920** [comment]
These changes breaks build rpm task in Wercker: https://app.wercker.com/#buildstep/56b947dc7929bc7d4e6bbb04

Could you fix it?
Perhaps, it will be buildable when ignoring `create-db.sql` in hatohol.spec.

File: hatohol.spec.in [hunk 1]
```diff
@@ -253,9 +253,10 @@ rm -rf %{buildroot}
 %{_unitdir}/hatohol.service
 %{_tmpfilesdir}/hatohol.conf
 %endif
-%{_datadir}/hatohol/sql/create-db.sql
```
Comment (cosmo0920) [inline hatohol.spec.in]: `%exclude %{_datadir}/hatohol/sql/create-db.sql` or something is needed here.

Comment (Mnakagawa) [inline hatohol.spec.in]: Thank you!
I fixed it.

**Comment by cosmo0920** [comment]
Looks good!
After CIs finishing, :shipit: !

**Comment by Mnakagawa** [comment]
Some tests ware failed. But, almost tests were successed.

File: hatohol.spec.in (hunk 1 above)
Comment (kz0817) [inline hatohol.spec.in]: Now 'create-db.sql' is being used only for test (TravisCI) from launch-hatohol-for-test.sh.
Why don't we stop installing it ?

I.e.  
The following lines are added in server/data/Makefile.am
<pre>
diff --git a/server/data/Makefile.am b/server/data/Makefile.am
index b22ebf6..c190228 100644
--- a/server/data/Makefile.am
+++ b/server/data/Makefile.am
@@ -1,5 +1,4 @@
 sql_DATA = \
-       create-db.sql \
        init-user.sql \
        server-type-zabbix.sql \
        server-type-nagios.sql \
@@ -14,6 +13,8 @@ sql_DATA = \
        incident-trackers.sql \
        actions.sql
 
+noinst_DATA = create-db.sql
+
 sqldir = $(pkgdatadir)/sql
 
 HATOHOL_SERVER_TYPE_UTIL = ../tools/hatohol-server-type-util
@@ -35,7 +36,8 @@ pkgsysconfdir = $(sysconfdir)/$(PACKAGE)
 
 EXTRA_DIST = \
        $(pkgsysconf_DATA) \
-       $(sql_DATA)
+       $(sql_DATA) \
+       $(noinst_DATA)
 
 CLEANFILES = \
        server-type-zabbix.sql \
</pre>
So we don't have to add %exclude in spec. file.

Comment (Mnakagawa) [inline hatohol.spec.in]: You are right.
I will send new PR

File: server/tools/hatohol-db-initiator.in [hunk 2]
```diff
@@ -235,14 +202,17 @@ def get_default_sql_dir():
...
 
 if __name__ == '__main__':
     init_config_map()
 
+    default_sql_dir = get_default_sql_dir()
+    if "--sql-dir" in sys.argv:
+        target_dir = sys.argv[sys.argv.index("--sql-dir")+1]
+    else:
+        target_dir = default_sql_dir
+    target_list = os.listdir(target_dir)
```
Comment (kz0817) [inline server/tools/hatohol-db-initiator.in]: This line should be in the 'if args.show_target' block.
os.listdir() is always invoked even when it's unnecessary.

File: server/tools/hatohol-db-initiator.in [hunk 3]
```diff
@@ -274,8 +244,19 @@ if __name__ == '__main__':
     parser.add_argument('-f', '--force', action='store_true',
                         help='Delete existing data in a table before the initial data are inserted. '
                         'By default, this tool skips to insert data if the table is not empty.')
-    parser.add_argument('-t', '--target', choices=get_target_list(),
-                        help='Only the specified table is initialized.')
+    parser.add_argument('-t', '--target',
+                        help='Only the specified table is initialized. '
+                        'Please check targets by executing "$hatohol-db-initiator --show-target"')
+    parser.add_argument('--show-target', action="store_true",
```
Comment (kz0817) [inline server/tools/hatohol-db-initiator.in]: I like plural in this context. I.e. --show-targets.
Or --target-list ?

File: server/tools/hatohol-db-initiator.in [hunk 4]
```diff
@@ -274,8 +244,19 @@ if __name__ == '__main__':
     parser.add_argument('-f', '--force', action='store_true',
                         help='Delete existing data in a table before the initial data are inserted. '
                         'By default, this tool skips to insert data if the table is not empty.')
-    parser.add_argument('-t', '--target', choices=get_target_list(),
-                        help='Only the specified table is initialized.')
+    parser.add_argument('-t', '--target',
+                        help='Only the specified table is initialized. '
+                        'Please check targets by executing "$hatohol-db-initiator --show-target"')
```
Comment (kz0817) [inline server/tools/hatohol-db-initiator.in]: targets ->
available targets

Omitting 'executing' looks more simpler. (Of course, it can be.)

**Pull request closed without merging**